demo/default.nix

```diff
@@ -1,6 +1,23 @@
 # A shell setup providing build tools and utilities for the demo
 let
   project = import ../default.nix { };
+  nix-thunk = import ./dep/nix-thunk { inherit (project) pkgs; };
+
+  deps = nix-thunk.mapSubdirectories nix-thunk.thunkSource ./dep;
+
+  pythonPackages = pkgs.python3Packages;
+
+  libtmux = pythonPackages.buildPythonPackage {
+    pname = "libtmux";
+    version = "0.15.0a1";
```

Comment:
What was the error you saw? Maybe the `libtmux` on a newer `nixpkgs` pin is sufficient instead of this, more manual, nix expression pinning?

Reply:
The error was a TypeError:
‘NoneType’ Object is Not Subscriptable in libtmux. Bumping to a new
version fixed the issue.

Reply:
A quick google yields this: https://github.com/tmux-python/tmuxp/issues/547. Are you on MacOS and maybe the nix-based `libtmux` version interacts with your host system? Can you try using a `--pure` nix shell?